tests/test_endpoint.py

```diff
@@ -134,77 +136,57 @@ def test_required_headers(self):
         foo = apiron.JsonEndpoint()
         assert {"Accept": "application/json"} == foo.required_headers
 
+    def test_str_method(self):
+        foo = apiron.JsonEndpoint(path="/bar/baz")
+        assert str(foo) == "/bar/baz"
+
+    def test_repr_method(self):
+        foo = apiron.JsonEndpoint(path="/bar/baz")
+        assert repr(foo) == "JsonEndpoint(path='/bar/baz')"
+
 
 class TestStreamingEndpoint:
     def test_format_response(self):
         foo = apiron.StreamingEndpoint()
         mock_response = mock.Mock()
         assert mock_response.iter_content(chunk_size=None) == foo.format_response(mock_response)
 
+    def test_str_method(self):
+        foo = apiron.StreamingEndpoint(path="/bar/baz")
+        assert str(foo) == "/bar/baz"
+
+    def test_repr_method(self):
+        foo = apiron.StreamingEndpoint(path="/bar/baz")
+        assert repr(foo) == "StreamingEndpoint(path='/bar/baz')"
+
 
 class TestStubEndpoint:
-    def test_stub_response(self):
-        """
-        Test initializing a stub endpoint with a stub response
-        """
-        stub_endpoint = apiron.StubEndpoint(stub_response="stub response")
-        assert "stub response" == stub_endpoint.stub_response
-
-    def test_extra_params(self):
-        """
-        Test initializing a stub endpoint with extra params
-        """
-        stub_endpoint = apiron.StubEndpoint(
-            stub_response="stub response",
-            path="/some/path/",
-            default_params={"param_name": "param_val"},
-            required_params={"param_name"},
-            arbitrary_kwarg="foo",
-        )
-        expected_params = {
-            "path": "/some/path/",
-            "default_params": {"param_name": "param_val"},
-            "required_params": {"param_name"},
-            "arbitrary_kwarg": "foo",
-        }
-        assert expected_params == stub_endpoint.endpoint_params
+    def test_stub_default_response(self, service):
+        service.stub_endpoint = apiron.StubEndpoint()
+        assert service.stub_endpoint() == {"response": "StubEndpoint(path='/')"}
 
     def test_call_static(self, service):
-        """
-        Test calling a ``StubEndpoint`` with a static response
-        """
         service.stub_endpoint = apiron.StubEndpoint(stub_response="stub response")
-        actual_response = service.stub_endpoint()
-        expected_response = "stub response"
+        assert service.stub_endpoint() == "stub response"
+
+    @pytest.mark.parametrize(
+        "test_call_kwargs,expected_response",
+        [({}, {"default": "response"}), ({"params": {"param_key": "param_value"}}, {"stub response": "stubby!"})],
+    )
```

Comment:
Separating test cases might enhance readability.

```python
    @pytest.mark.parametrize(
        "test_call_kwargs,expected_response", [
            ({}, {"default": "response"}),
            ({"params": {"param_key": "param_value"}}, {"stub response": "stubby!"})
        ],
    )
```

I like `@pytest.mark.parametrize`, but I think it's a good idea to take extra care to preserve readability and clarity.
```

Reply:
I think `black` did this for me, because I did have them on separate lines. Maybe in the future we can stop blackening tests?

Reply:
Black did that? I'm alarmed. It's a big sacrifice to stop linting tests. Seems like a last resort. I'd switch linters first.

Reply:
Once a list grows long enough to span more than one line's worth, it splits them all out. I don't necessarily agree with that decision, but I suppose the spirit of `black` is that it makes such discussions moot 😄